tests/util/block.ts

```diff
@@ -222,6 +224,37 @@ export const verifyBlockFees = async (
                 let feePortions = calculateFeePortions(fee.partialFee.toBigInt());
                 txFees = fee.partialFee.toBigInt();
                 txBurnt += feePortions.burnt;
+
+                // verify entire substrate txn fee
+                // let apiAt = await context.polkadotApi.at(blockDetails.block.hash.toString());
+                let apiAt = await context.polkadotApi.at(previousBlockHash); // TODO: i think this is right because multiplier is updated in on_finalize
+                let lengthFee = (await apiAt.call.transactionPaymentApi.queryLengthToFee(dispatchInfo.encodedLength) as any).toBigInt();
+
+                let unadjustedWeightFee = (await apiAt.call.transactionPaymentApi.queryWeightToFee(dispatchInfo.weight) as any).toBigInt();
```

Comment:
Something seems very broken about `dispatchInfo.weight`. In this particular case (running test `"should be able to calculate entire fee"`), it's reporting a `refTime` of `423314000` which is definitely wrong.

The real value should be `173314000`, which is correct in `fee.weight`. Why is `dispatchInfo` so wrong?

Reply:
Furthermore, why is it used [presumably] correctly (above in the same fn) for Ethereum txns?